components/server/src/ide-service.ts

```diff
@@ -67,12 +67,23 @@ export class IDEService {
         }
     }
 
+    async isIDEAvailable(ide: string, request: { user: { id: string; email?: string } }): Promise<boolean> {
```

Comment:
nit: no need to specify the email here
```suggestion
    async isIDEAvailable(ide: string, request: { user: { id: string } }): Promise<boolean> {
```

Reply:
This is a copy of `getIDEConfig` 's param